sshproxy.go

```diff
@@ -307,6 +309,10 @@ func (s *Server) launchCommand(channel ssh.Channel, cmd *exec.Cmd, terminal *pty
 		}
 	}
 
+	if s.shellCreds != nil {
+		cmd.SysProcAttr = &syscall.SysProcAttr{Credential: s.shellCreds}
```

Comment:
`resin-git` needs to spawn the shell process as a different user, setting `SysProcAttr.Credential` was the only way I could see of achieving this without messing about using `sudo`.